frontend/src/app/main/data/workspace/persistence.cljs

```diff
@@ -376,22 +409,54 @@
               {:file-id file-id
                :is-local local?
                :url uri
-               :name name})]
+               :name (or name (url-name uri))})
+
+            file-stream
+            (when data
+              (->> (rx/from data)
+                   (rx/map di/validate-file)))
+
+            image-stream
+            (cond
+              (seq uris)
+
+              (rx/merge
+               (->> (rx/from uris)
+                    (rx/filter (comp not svg-url?))
+                    (rx/map prepare-uri)
+                    (rx/mapcat #(rp/mutation! :create-file-media-object-from-url %))
+                    (rx/do on-image))
+
+               (->> (rx/from uris)
+                    (rx/filter svg-url?)
+                    (rx/merge-map fetch-svg)
+                    (rx/merge-map parse-svg)
+                    (rx/with-latest vector uris)
+                    (rx/map #(assoc (first %) :name (or name (url-name (second %)))))
+                    (rx/do on-svg)))
```

Comment:
As far as I understand it, this stream is just for "side-effect" (executing the `do`). In this case I think the `rx/ignore` is missing here. This will avoid emiting the result of this stream to the STORE stream.